back/engines/commercial/report_builder/app/services/report_builder/queries/projects.rb

```diff
@@ -0,0 +1,67 @@
+module ReportBuilder
+  class Queries::Projects < ReportBuilder::Queries::Base
+    def run_query(start_at: nil, end_at: nil, **_other_props)
+      start_date, end_date = TimeBoundariesParser.new(start_at, end_at).parse
+
+      projects_that_have_phases_ids = Phase.select(:project_id)
```

Comment:
Unfortunately not- you can create a project, but not create any phases in it

Reply:
Yeah your refactored version of the code already handles this case correctly during the

```
overlapping_project_ids = Phase
        .select(:project_id)
...
```

part. This will only include projects that have phases.